enhancements/compliance-operator.md

```diff
@@ -0,0 +1,222 @@
+to asses compliance against a given standard (for example FedRAMP moderate)
+and in a future version, remediate the findings to get the cluster into
+a compliant state. The scan is performed by the OpenSCAP tool.
+OpenSCAP is well-established, [standards compliant](https://www.open-scap.org/features/standards/)
+and [NIST-certified](https://csrc.nist.gov/projects/security-content-automation-protocol-validation-pr/validated-products-and-modules/142-red-hat-scap-1-2-product-validation-record)
+project that administrators are aware of and trust.
+
+For the first iteration, the operator would only provide a report with
+the findings and proposed remediations which the cluster administrator
+can then apply on their own. Remediation would be developed in one
+of the future versions.
+
+## Motivation
+
+The default installation of OKD is not compliant (and is not expected
+to be, as some requirements have impact on performance or flexibility of use
+of the cluster) with different standards users in regulated environments
+must adhere to.
+
+Enabling users to be compliant with standards their respective industry
+requires would enable OKD to be used in those environments. This is especially
+true of the US public sector.
+
+### Dependencies
+As said above, the operator would orchestrate OpenSCAP scans on cluster
+nodes. However, there are several areas where OpenSCAP needs to be extended
+to be better usable for scanning an OKD cluster:
+
+* OpenSCAP did not support merging of results from several nodes into a single
+  bundle. Each scan on each node produces a separate report. There is an
+  RFE filed against OpenSCAP and planned by the OpenSCAP team to aggregate
+  results from all the cluster nodes into a single report.
+* The remediations were typically done using a shell script or Ansible. For
+  OpenShift we’d want the remediations done using MachineConfigs or other
+  Custom Resources. However, the remediations are not a hard prerequisite for
+  the first version of the operator, even identifying the gaps in compliance
+  would be useful.
+* For the gaps in compliance we identify, a check must be written. These
+  checks are being compiled as a parallel effort. Having a full set of
+  checks and/or remediations is not a hard prerequisite for the operator
+  itself, though as the content would be delivered separately.
+
+These have all been filed with the OpenSCAP team and acknowledged by them.
+
+### Goals
+
+A cluster administrator is able to use the operator to assess the degree
+of compliance against a given security standard. In particular this would
+mean:
+
+* Provide a way for cluster administrator to start cluster scans
+* Inspect the scan result on the high level (pass/fail/error)
+* Provide a way to collect the detailed results of the scan
+  * The results would list a way to remediate the gaps, typically by the means
+    of deploying certain manifests to OKD (e.g. a Machineconfig), a different
```

Comment:
s/OKD/OpenShift/ ?

Reply:
Probably? :)

Just to get the terminology right, OKD is used for the Kubernetes piece only and OpenShift = OKD + RHCOS?